### ISSUE TITLE
v2ray-plugin breaks SS clients that do not support it

### ISSUE THREAD

**Issue by hjiang** [StreisandEffect/streisand#1577]
From the generated docs, `v2ray-plugin` should be optional, since it is only mentioned in the last section of the ShadowSocks instructions page. But after I set up a fresh Streisand instance, ShadowSocks did not work out of box. I tested on Surge (macOS and iOS) and Potatso LIte (iOS). Surge reported decryption error. After I removed the two plugin lines from `/etc/shadowsocks-libev/config.json` and restarted shadowsocks-libev, all the clients work correctly.

**Comment by akakoori** [comment]
Same here. 
It seems that even the latest version of shadowsocksX-NG **DOES NOT** implement v2ray-plugin by default:

![image](https://user-images.githubusercontent.com/15867066/58366669-5be96880-7f08-11e9-9700-f59558a5f9ed.png)

Thus, there are two solutions:
+ On your server
1. Disable v2ray-plugin 
2. Restart shadowsocks-libev. 
Not recommended as this is not safe.
+ On your client
1. Manually download (and build if you clone it from github) v2ray-plugin from [here](https://github.com/shadowsocks/v2ray-plugin) 
2. Rename it as `v2ray-plugin`
3. Move it to `~/Library/Application Support/ShadowsocksX-NG/v2ray-plugin_master/v2ray-plugin`
4. Edit the setting of your server in ShadowsocksX-NG as follow:
![image](https://user-images.githubusercontent.com/15867066/58366900-3a3db080-7f0b-11e9-83f1-cf2c65a33969.png)
5. Hooray!

This issue probably wont fix until ShadowsocksX-NG released a new version embed with v2ray-plugin.
However the doc should be updated.

**Comment by hjiang** [comment]
Most mobile clients probably wont support v2ray-plugin due to the client-side complexity. I think it's a good idea to make it an option during installation (disabled by default), or make vanilla shadowsocks available on another port.

**Comment by B451YbUz75Rn** [comment]
为什么你的使用会这么复杂，手机端安装shadowckes, 去谷歌应用商店下载 v2ray-plugin （此程序是个插件）无法单独运行，在你的shadowckes 【配置文件设置】最下【插件】选择【v2ray】 即可连接。在不使用插件的情况下shadowckes 是无法使用的，即便你能链接成功。

**Comment by akakoori** [comment]
@B451YbUz75Rn Google App Store is not available on iOS.

**Comment by hjiang** [comment]
Most clients do not support v2ray-plugin. And the documentation is certainly wrong.

**Comment by denoza** [comment]
@hjiang I disagree, as per my information only iOS clients have this issue. The official windows, linux and android clients all have `v2ray-plugin` support. However, your suggestion is right, we can either install  vanilla `shadowsocks` by default and then provide the option to install `v2ray-plugin` or we can update the documentation. The `simple-obfs` plugin installed previously did not have this issue. And we actually dropped the idea of using `failover` option with it due to the similar problems as this issue.

**Comment by hjiang** [comment]
@denoza I'm only using Shadowsocks on Apple OSes, and it seems I happened to only tried the clients that don't have v2ray support, so sorry about making the assertion about "most clients". Yes, `simple-obfs` is more widely supported in my experience as well.

**Comment by thiras** [comment]
I confirm freshly installed shadowsocks doesn't work without the plugin. This is big mistake. It's the easiest method streisand provides and should be widely available.

**Comment by nabilfreeman** [comment]
I ran into this problem today. Had a feeling it was v2ray as I noticed it in the Ansible logs. The latest ShadowsocksX-NG doesn't support v2ray let alone Potatso etc for iOS.

So this essentially makes Streisand via Shadowsocks useless for anything but personal use in specific situations.

**Comment by hadifarnoud** [comment]
V2ray is awesome. This is the best thing to fight censorship in countries like China and Iran. 

I wasn't able to get it to work though. I did what @akakoori said with no luck. Also tried on iOS with Shadowrocket app but there are some parameters on this app that I don't know. Can someone help me with v2ray settings on iOS?
![IMG_685EEE8962C2-1](https://user-images.githubusercontent.com/589808/61639795-64b0ad80-acb1-11e9-8904-8bd81db325cf.jpeg)

**Comment by NamTaf** [comment]
@hadidarnoud enter your server address and port (8530) in the fields shown in your screenshot, then Shadowrocket should work.

Also, for the Windows shadowsocks client, I had luck downloading the latest release off the shadowsocks github (https://github.com/shadowsocks/v2ray-plugin/releases), renaming it to ‘v2ray-plugin.exe’ and selecting the appropriate settings in shadowsocks client v4.1.7.1. Both now work fine with v2ray-plugin selected on my Streisand server.

**Comment by ortonomy** [comment]
New Shadowsocks-X-NG supports v2ray plugin, but I can't get it working with my streisand server. Anyone know how to configure v2ray for shadowsocks-X-NG for Streisand?
